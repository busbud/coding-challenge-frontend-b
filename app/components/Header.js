var React = require('react');

// Stateless Functional component - since it only has a render method
function Header () {
	return (
		<header className='title'>
			<img 
				src={require('../images/osheaga_logo.png')}
				alt='Osheaga Logo'
			/>
		</header>
	);
}

module.exports = Header;