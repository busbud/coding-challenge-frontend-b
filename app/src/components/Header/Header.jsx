// React
import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
import { injectIntl } from "react-intl";
// nodejs library that concatenates classes
import classNames from "classnames";
// Material-UI HigherOrderComponent styling
import withStyles from "@material-ui/core/styles/withStyles";
// Material-UI Core Components
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Hidden from "@material-ui/core/Hidden";
import Drawer from "@material-ui/core/Drawer";
// @material-ui/icons
import Menu from "@material-ui/icons/Menu";
import Close from "@material-ui/icons/Close";
import Brand from "../Brand/Brand";
// Logo
import logo from "../../logo.svg";
// Styles
import headerStyle from "../../assets/jss/components/headerStyle";

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      mobileOpen: false,
    };
    this.handleDrawerToggle = this.handleDrawerToggle.bind(this);
  }

  handleDrawerToggle() {
    this.setState({ mobileOpen: !this.state.mobileOpen });
  }

  render() {
    const { classes, links, intl } = this.props;
    const title = intl.formatMessage({ id: "title" });

    return (
      <AppBar
        className={classNames(classes.appBar, classes.absolute, classes.fixed)}
      >
        <Toolbar className={classes.container}>
          <Brand logo={logo} logoAltText={title} />
          <Hidden smDown implementation="css" className={classes.hidden}>
            <div className={classes.collapse}>{links}</div>
          </Hidden>
          <Hidden mdUp>
            <IconButton
              color="inherit"
              aria-label="open drawer"
              onClick={this.handleDrawerToggle}
            >
              <Menu />
            </IconButton>
          </Hidden>
        </Toolbar>
        <Hidden mdUp implementation="css">
          <Drawer
            variant="temporary"
            anchor={"right"}
            open={this.state.mobileOpen}
            classes={{
              paper: classes.drawerPaper,
            }}
            onClose={this.handleDrawerToggle}
          >
            <IconButton
              color="inherit"
              aria-label="open drawer"
              onClick={this.handleDrawerToggle}
              className={classes.closeButtonDrawer}
            >
              <Close />
            </IconButton>
            <div className={classes.appResponsive}>{links}</div>
          </Drawer>
        </Hidden>
      </AppBar>
    );
  }
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
  links: PropTypes.node.isRequired,
  name: PropTypes.string,
  brand: PropTypes.node,
};

export default injectIntl(withStyles(headerStyle)(Header));
