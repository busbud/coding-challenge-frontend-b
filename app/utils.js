// UTC time (dateTtime) -> hh:mm
export const utcTimeToHhMm = utcTime => utcTime.split('T')[1].substr(0, 5)

// Number XXXX -> XX,XX (currency)
export const numToCurrency = num => (num/100).toFixed(2)
