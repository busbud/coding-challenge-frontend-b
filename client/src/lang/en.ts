export default {
  title: "Find bus tickets for your trip",
  chooseCurrency: "Choose Currency",
  chooseLanguage: "Choose Language",
  origin: "Origin",
  destination: "Destination",
  date: "Date",
  passenger: "Passenger",
  submit: "Submit",
  passengerUnit: "passenger",
  resultList: "Result List",
};
