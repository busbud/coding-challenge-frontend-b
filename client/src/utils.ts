export const toCamel = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace("-", "").replace("_", "");
  });
};

export const capitalize = (str: string) => {
  const lower = str.toLowerCase();
  return str.charAt(0).toUpperCase() + lower.slice(1);
};
