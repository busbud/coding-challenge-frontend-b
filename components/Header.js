import PropTypes from "prop-types";

import { i18n, withTranslation } from "../i18n";
import { colours, fonts } from "../theme";

const Header = props => (
  <header>
    <div className="locale-buttons-wrapper">
      <button
        className="locale-button"
        type="button"
        onClick={() => i18n.changeLanguage("en")}
      >
        en
      </button>
      <button
        className="locale-button"
        type="button"
        onClick={() => i18n.changeLanguage("fr")}
      >
        fr
      </button>
    </div>
    <div className="background">
      <img className="background-image" src="/static/img/background.jpg" />
    </div>
    <div className="banner">
      <img
        className="banner-image"
        src="/static/img/banner.png"
        alt="Osheaga Festival Musique Et Arts"
      />
    </div>
    <h1>{props.t("heading")}</h1>
    {props.children}
    <style jsx>{`
      header {
        color: white;
        padding-bottom: 64px;
        padding-left: 24px;
        padding-right: 24px;
        padding-top: 24px;
        position: relative;
      }
      .locale-buttons-wrapper {
        margin-bottom: 32px;
        margin-left: auto;
        margin-right: auto;
        max-width: 960px;
        text-align: right;
      }
      .locale-button {
        background-color: ${colours.blueLight};
        color: ${colours.blue};
        cursor: pointer;
        font-family: ${fonts.slab};
        font-size: 1rem;
        margin-left: 8px;
        width: 48px;
      }
      .background {
        height: 100%;
        left: 0;
        position: absolute;
        top: 0;
        width: 100%;
        z-index: -1;
      }
      .background-image {
        height: inherit;
        object-fit: cover;
        width: 100%;
      }
      .banner {
        margin-left: auto;
        margin-right: auto;
        max-width: 400px;
      }
      .banner-image {
        height: auto;
        max-width: 100%;
      }
      h1 {
        color: white;
        font-size: 3rem;
        line-height: 1em;
        margin-bottom: 0;
        margin-top: 32px;
        text-align: center;
        text-shadow: 1px 1px 4px ${colours.blue};
      }
    `}</style>
  </header>
);

Header.propTypes = {
  children: PropTypes.node,
  t: PropTypes.func.isRequired
};

export default withTranslation("common")(Header);
