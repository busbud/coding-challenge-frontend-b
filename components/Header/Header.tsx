import Link from 'next/link'
import { useRouter } from 'next/router'

import { Locale } from '../../types'

export default function Header() {
  const router = useRouter()
  const locale: Locale = router.locale as Locale

  return (
    <header className="app-header">
      <h1 className="logo">Bus To Fest</h1>

      <div className="locale-links">
        { locale !== 'en' && <Link href="/" locale="en">EN</Link>}
        { locale !== 'fr' && <Link href="/" locale="fr">FR</Link>}
      </div>
    </header>
  )
}