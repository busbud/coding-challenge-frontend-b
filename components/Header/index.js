import React, { useContext } from "react";
import styled from "styled-components";

import { FullWidth, SiteWidth } from "../common-styled/Containers";
import { IntlContext } from "../../pages/_app";
import { getTransaltion } from "../../utils/translation";

const Wrapper = styled(FullWidth)`
  background: #1da1f2;
  align-items: center;
`;
const InnerWrapper = styled(SiteWidth)`
  align-items: center;
  flex-direction: row;
`;
const Title = styled.h1`
  font-size: 50px;
  margin: 0;
  color: ${({ theme }) => theme.text.white};
`;
const Languages = styled.div`
  margin-left: auto;
  display: flex;
  align-items: center;
`;
const Language = styled.a`
  padding: 10px;
  color: #fff;
  cursor: pointer;
`;

export default function Header(props) {
  const { language, toggleLanguage } = useContext(IntlContext);
  return (
    <Wrapper>
      <InnerWrapper>
        <Title data-testid="header-title">{props.title}</Title>
        <Languages>
          <Language
            data-testid="header-english"
            title={getTransaltion("header.enTitle", language)}
            role="button"
            onClick={() => toggleLanguage("en")}
          >
            English
          </Language>
          <Language
            data-testid="header-french"
            title={getTransaltion("header.frTitle", language)}
            role="button"
            onClick={() => toggleLanguage("fr")}
          >
            French
          </Language>
        </Languages>
      </InnerWrapper>
    </Wrapper>
  );
}
