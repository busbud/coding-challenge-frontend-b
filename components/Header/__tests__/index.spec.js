import React from "react";
import { fireEvent } from "@testing-library/react";

import Header from "../index";
import { setupComponent } from "../../../utils/jestUtils";
import en from "../../../translations/en";

describe("Header component", () => {
  it("should render without throwing an error", function() {
    const { asFragment } = setupComponent(<Header title={en.siteName} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render title received in props", function() {
    const { getByTestId } = setupComponent(<Header title={en.siteName} />);
    expect(getByTestId("header-title")).toHaveTextContent(en.siteName);
  });

  it("should have all the required attributes for language selection", function() {
    const { getByTestId } = setupComponent(<Header title={en.siteName} />);
    expect(getByTestId("header-english")).toHaveAttribute("role");
    expect(getByTestId("header-english")).toHaveAttribute("title");

    expect(getByTestId("header-french")).toHaveAttribute("role");
    expect(getByTestId("header-french")).toHaveAttribute("title");
  });

  it("should call toggle language", async function() {
    const toggle = jest.fn();
    const { getByTestId } = setupComponent(
      <Header title={en.siteName} />,
      toggle
    );
    fireEvent.click(getByTestId("header-french"));
    expect(toggle).toHaveBeenCalled();

    fireEvent.click(getByTestId("header-english"));
    expect(toggle).toHaveBeenCalled();
  });
});
