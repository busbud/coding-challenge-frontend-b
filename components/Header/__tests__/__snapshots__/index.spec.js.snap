// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render without throwing an error 1`] = `
<DocumentFragment>
  <section
    class="sc-bwzfXH sc-bxivhb gCAUCy"
  >
    <div
      class="sc-htpNat sc-ifAKCX hxabGC"
    >
      <h1
        class="sc-EHOje gieLlh"
        data-testid="header-title"
      >
        Book a ride
      </h1>
      <div
        class="sc-bZQynM gINrGK"
      >
        <a
          class="sc-gzVnrw bmBGcZ"
          data-testid="header-english"
          role="button"
          title=""
        >
          English
        </a>
        <a
          class="sc-gzVnrw bmBGcZ"
          data-testid="header-french"
          role="button"
          title=""
        >
          French
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
